File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/ReleaseLocationUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.command.ReleaseLocationCommand;
import com.dozycoffee.wms.warehouse.application.port.in.result.LocationResult;
import reactor.core.publisher.Mono;

public interface ReleaseLocationUseCase {

    Mono<LocationResult> release(ReleaseLocationCommand command);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/ReleaseLocationUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.command.ReleaseLocationCommand
import com.dozycoffee.wms.warehouse.application.port.`in`.result.LocationResult
import reactor.core.publisher.Mono

interface ReleaseLocationUseCase {
    fun release(command: ReleaseLocationCommand): Mono<LocationResult>
}
